src/Microsoft.OData.Core/ODataReaderCore.cs

```diff
@@ -869,6 +883,31 @@ protected internal class Scope
             /// In all cases the specified type must be an structured type.</remarks>
             [SuppressMessage("Microsoft.Performance", "CA1800:DoNotCastUnnecessarily", Justification = "Debug.Assert check only.")]
             internal Scope(ODataReaderState state, ODataItem item, IEdmNavigationSource navigationSource, IEdmStructuredType expectedResourceType, ODataUri odataUri)

```

Comment:
>IEdmStructuredType expectedResourceType [](start = 106, length = 39)

maybe you can do like this:

Add a bool argument as "bool isNullable = true"

Reply:
what's that?

---
In reply to: [264344042](https://github.com/OData/odata.net/pull/1322#discussion_r264344042) [](ancestors = 264344042)

Reply:
I decided to add another constructor as I had an IsNullable optional parameter to begin with but we wanted to use type reference instead.

---
In reply to: [266614311](https://github.com/OData/odata.net/pull/1322#discussion_r266614311) [](ancestors = 266614311,264344042)